src/Featurizers/CountVectorizerFeaturizer.h

```diff
@@ -140,74 +126,20 @@ class CountVectorizerEstimatorImpl : public TransformerEstimator<std::string, Sp
     bool begin_training_impl(void) override;
 
     // MSVC has problems when the declaration and definition are separated
-    FitResult fit_impl(typename BaseType::InputType const *, size_t) override {
-        throw std::runtime_error("This should not be called");
+    FitResult fit_impl(typename BaseType::InputType const *pBuffer, size_t cElements) override {
+        return _tfidfEstimator.fit(pBuffer, cElements);
     }
 
     void complete_training_impl(void) override;
 
     // MSVC has problems when the declaration and definition are separated
     typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
-
-        using DocumentStatisticsEstimator = Components::DocumentStatisticsEstimator<MaxNumTrainingItemsV>;
-
-        Components::DocumentStatisticsAnnotationData const &      data(DocumentStatisticsEstimator::get_annotation_data(BaseType::get_column_annotations(), _colIndex, Components::DocumentStatisticsEstimatorName));
-
-        typename Components::DocumentStatisticsAnnotationData::FrequencyAndIndexMap const &
-                                                            freqAndIndexes(data.TermFrequencyAndIndex);
-        typename CountVectorizerTransformer::IndexMapType indexMap;
-
-        for (auto const & freqAndIndex : freqAndIndexes) {
-            indexMap.insert(std::make_pair(freqAndIndex.first, freqAndIndex.second.Index));
-        }
-
-        return typename BaseType::TransformerUniquePtr(new CountVectorizerTransformer(indexMap, _binary, _lower, _analyzer, _regex, _ngram_min, _ngram_max));
+        _tfidfEstimator.complete_training();
+        typename TfidfEstimator::TransformerUniquePtr           pTransformer(CountVectorizerEstimator<MaxNumTrainingItemsV>::_tfidfEstimator.create_transformer());
```

Comment:
>CountVectorizerEstimator<MaxNumTrainingItemsV>:: [](start = 77, length = 48)

I don't think that this part should be necessary; is it enough to call _tfidfEstimator?